src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,71 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String VALUE_CRITERIA = "supply";
+    private static final String LINE_SEPARATOR = " ";
+
+    public static void getStatistic(String fromFileName, String toFileName) {
+        List<Integer> listOfSupplyCosts = new ArrayList<>();
+        List<Integer> listOfBuyCosts = new ArrayList<>();
+        readFromFile(fromFileName, listOfSupplyCosts, listOfBuyCosts);
+        int sumOfSupplies = getSumOfList(listOfSupplyCosts);
+        int sumOfBuys = getSumOfList(listOfBuyCosts);
+        String result = createReport(sumOfSupplies, sumOfBuys, sumOfSupplies - sumOfBuys);
+        createAndWriteToFile(toFileName, result);
+    }
 
-    public void getStatistic(String fromFileName, String toFileName) {
+    public static void readFromFile(String fromFile, List<Integer> supplies, List<Integer> buys) {
```

Comment:
separate reading and counting logic int two methods, `readFromFile()` should only read data, not count it